sdk/core/core-rest-pipeline/src/policies/systemErrorRetryPolicy.ts

```diff
@@ -61,104 +42,8 @@ export interface SystemErrorRetryPolicyOptions {
-  const retryInterval = options.retryDelayInMs ?? DEFAULT_CLIENT_RETRY_INTERVAL;
-  const maxRetryInterval = options.maxRetryDelayInMs ?? DEFAULT_CLIENT_MAX_RETRY_INTERVAL;
-
-  function shouldRetry(retryData: RetryData, err?: RetryError): boolean {
-    if (!isSystemError(err)) {
-      return false;
-    }
-    const currentCount = retryData.retryCount;
-    return currentCount <= retryCount;
-  }
-
-  function updateRetryData(retryData: RetryData, err?: RetryError): RetryData {
-    if (err) {
-      if (retryData.error) {
-        err.innerError = retryData.error;
-      }
-
-      retryData.error = err;
-    }
-
-    // Adjust retry count
-    retryData.retryCount++;
-
-    // Exponentially increase the delay each time
-    const exponentialDelay = retryInterval * Math.pow(2, retryData.retryCount);
-    // Don't let the delay exceed the maximum
-    const clampedExponentialDelay = Math.min(maxRetryInterval, exponentialDelay);
-    // Allow the final value to have some "jitter" (within 50% of the delay size) so
-    // that retries across multiple clients don't occur simultaneously.
-    const delayWithJitter =
-      clampedExponentialDelay / 2 + getRandomIntegerInclusive(0, clampedExponentialDelay / 2);
-
-    retryData.retryInterval = delayWithJitter;
-
-    return retryData;
-  }
-
-  async function retry(
-    next: SendRequest,
-    retryData: RetryData,
-    request: PipelineRequest,
-    response?: PipelineResponse,
-    requestError?: RetryError
-  ): Promise<PipelineResponse> {
-    retryData = updateRetryData(retryData, requestError);
-    if (shouldRetry(retryData, requestError)) {
-      try {
-        logger.info(`Retrying request in ${retryData.retryInterval}`);
-        await delay(retryData.retryInterval);
-        const res = await next(request);
-        return retry(next, retryData, request, res);
-      } catch (e) {
-        return retry(next, retryData, request, response, e);
-      }
-    } else if (requestError || !response) {
-      // If the operation failed in the end, return all errors instead of just the last one
-      const err =
-        retryData.error ||
-        new RestError("Failed to send the request.", {
-          code: RestError.REQUEST_SEND_ERROR,
-          statusCode: response?.status,
-          request: response?.request,
-          response
-        });
-      throw err;
-    } else {
-      return response;
-    }
-  }
-
   return {
     name: systemErrorRetryPolicyName,
-    async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
-      const retryData = {
-        retryCount: 0,
-        retryInterval: 0
-      };
-      try {
-        const response = await next(request);
-        return retry(next, retryData, request, response);
-      } catch (e) {
-        const error: RestError = e;
-        return retry(next, retryData, request, error.response, error);
-      }
-    }
+    sendRequest: retryPolicy(exponentialRetryStrategy(options)).sendRequest
```

Comment:
this wouldn't support customizing max retries, would it? Since the strategy never updated `state.maxRetries`?

Reply:
Yes it allows customizing max retries since exponentialRetryStrategy is a function that uses the maxRetries received from the options to alter the state 🤔

Reply:
New approach! https://github.com/Azure/azure-sdk-for-js/pull/19078/commits/706de41ce49045e4afaa8559dafcd8204a1e122c